lib/internal/streams/compose.js

```diff
@@ -94,63 +121,119 @@ module.exports = function compose(...streams) {
   });
 
   if (writable) {
-    d._write = function(chunk, encoding, callback) {
-      if (head.write(chunk, encoding)) {
-        callback();
-      } else {
-        ondrain = callback;
-      }
-    };
-
-    d._final = function(callback) {
-      head.end();
-      onfinish = callback;
-    };
-
-    head.on('drain', function() {
-      if (ondrain) {
-        const cb = ondrain;
-        ondrain = null;
-        cb();
-      }
-    });
+    if (isNodeStream(head)) {
+      d._write = function(chunk, encoding, callback) {
+        if (head.write(chunk, encoding)) {
+          callback();
+        } else {
+          ondrain = callback;
+        }
+      };
 
-    tail.on('finish', function() {
-      if (onfinish) {
-        const cb = onfinish;
-        onfinish = null;
-        cb();
-      }
-    });
+      d._final = function(callback) {
+        head.end();
+        onfinish = callback;
+      };
+
+      head.on('drain', function() {
+        if (ondrain) {
+          const cb = ondrain;
+          ondrain = null;
+          cb();
+        }
+      });
+    } else if (isWebStream(head)) {
+      const writable = isTransformStream(head) ? head.writable : head;
+      const writer = writable.getWriter();
+
+      d._write = async function(chunk, encoding, callback) {
+        try {
+          await writer.ready;
+          writer.write(chunk).catch(() => {});
```

Comment:
Any reason we suppress the error here and bellow?

Reply:
Like, can you explain why it's not awaited instead of the empty catch?

Reply:
Same as the explanation here https://streams.spec.whatwg.org/#example-manual-write-with-backpressure, 
awaiting writer.ready is good for both catching errors and backpressure making awaiting writer.write redundant and empty catch helps avoid any unhandled rejections

Reply:
also awaiting would make us wait until the write is finished, we can simply enqueue and move on errors if any would be caught by writer.ready

Reply:
Should I mark these resolved? @benjamingr ?